test_ros2cli/test/test_cli.py.in

```diff
@@ -0,0 +1,139 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+
+import unittest
+
+from launch import LaunchDescription
+from launch.actions import ExecuteProcess
+from launch.actions import OpaqueFunction
+from launch.actions import TimerAction
+from launch.substitutions import LaunchConfiguration
+
+import launch_testing
+import launch_testing.asserts
+import launch_testing.util
+
+sys.path.append(os.path.dirname(__file__))
+
+from @TEST_CONFIG_MODULE@ import get_test_configurations  # noqa
+
+
+@launch_testing.parametrize(
+    'test_configuration', get_test_configurations(
+        rmw_implementation='@TEST_RMW_IMPLEMENTATION@'
+    )
+)
+def generate_test_description(test_configuration, ready_fn):
+    cmd = ['ros2', test_configuration.command]
+    cmd.extend(test_configuration.arguments)
+    command_under_test = ExecuteProcess(
+        cmd=cmd,
+        name='cli',
+        output='screen',
+        additional_env={'PYTHONUNBUFFERED': '1'},
+        sigterm_timeout=LaunchConfiguration(
+            'sigterm_timeout', default=60
+        )
+    )
+    actions = [command_under_test]
+    actions.append(
+        OpaqueFunction(function=lambda context: ready_fn())
+    )
+    launch_description = LaunchDescription([
+        # Always restart daemon to isolate tests.
```

Comment:
Maybe instead of doing this it would be simpler to add an option `--no-daemon` to all the commands to not start the daemon on demand?

Reply:
Hmm, that is a good idea, but won't it result in test instances affecting each other through daemon state?